languages/go/query.go

```diff
@@ -249,25 +239,37 @@ func (q Query) handleExternalOp(event types.QueryEventExternalOp) error {
 	if err != nil {
 		return err
 	}
+
 	var answer bool
-	leftCmp := left.(interfaces.Comparer)
-	rightCmp := right.(interfaces.Comparer)
-	// @TODO: Where are the implementations for these for builtin stuff (numbers mainly)
-	switch event.Operator.OperatorVariant.(type) {
-	case OperatorLt:
-		answer = leftCmp.Lt(rightCmp)
-	case OperatorLeq:
-		answer = leftCmp.Lt(rightCmp) || leftCmp.Equal(rightCmp)
-	case OperatorGt:
-		answer = rightCmp.Lt(leftCmp)
-	case OperatorGeq:
-		answer = !leftCmp.Lt(rightCmp)
-	case OperatorEq:
-		answer = leftCmp.Equal(rightCmp)
-	case OperatorNeq:
-		answer = !leftCmp.Equal(rightCmp)
-	default:
-		return fmt.Errorf("Unsupported operation: %v", event.Operator.OperatorVariant)
+
+	leftCmp, leftOk := left.(interfaces.Comparer)
```

Comment:
I think we want to make this so that if _either_ the left or the right implements `Comparer`, it can use that implementation of `Comparer`. And update the `Comparer` interface itself so that it takes in `interface{}` as the argument.

That way you can implement comparer for typedefs in Go in a sensible way. There's a little more information in https://github.com/osohq/oso/issues/795#issue-838921051 but the short of it is, IMO the ideal Go interface would be:

```go
type Comparer interface {
	// Should return `true` when values are equal; `false` when not equal.
	Equal(other interface{}) bool
	// Should return `true` when the comparer is less than `other`; `false` when not less than.
	Lt(other interface{}) bool
}
```

User implements:
```go
type MyString string

func (left MyString ) Equal(right interface{}) bool {
    if other, ok := right.(string); ok    {
		return left == other
	}
}
```

It needs to check that at least one of the two operands implementing Comparer, because you might end up with: `new MyString("abc") == "abc"` or you might have `"abc" == new MyString("abc")` and both should work without needing to implement `Comparer` for string

Reply:
👍 done, check it out; it makes things a little more complicated because you can't just negate opposite order relations when x and y might have totally different types.